sdks/java/io/text/src/test/java/org/apache/beam/sdk/io/text/TextIOIT.java

```diff
@@ -0,0 +1,126 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.text;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.text.ParseException;
+import java.util.Date;
+
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.io.common.HashingFn;
+import org.apache.beam.sdk.io.common.IOTestPipelineOptions;
+import org.apache.beam.sdk.io.common.TestTextLine;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+/**
+ * A test of {@link org.apache.beam.sdk.io.TextIO}.
+ *
+ * <p>Run this test using the command below. Pass in connection information via PipelineOptions:
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/text -DintegrationTestPipelineOptions='[
+ *  "--linesOfTextCount=100000" ]'
+ * </pre>
+ *
+ * <p>For now, this test can be run on DirectRunner only.
+ * */
+@RunWith(JUnit4.class)
+public class TextIOIT {
+
+  private static String filenamePrefix;
+  private static Long linesOfTextCount;
+
+  @Rule
+  public TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setup() throws ParseException {
+    PipelineOptionsFactory.register(IOTestPipelineOptions.class);
+    IOTestPipelineOptions options = TestPipeline.testingPipelineOptions()
+        .as(IOTestPipelineOptions.class);
+
+    if (options.getLinesOfTextCount() != null) {
+      linesOfTextCount = options.getLinesOfTextCount();
+    }
+
+    filenamePrefix = generateFileBasename();
+  }
+
+  private static String generateFileBasename() {
+    return String.format("TEXTIOIT_%s", new Date().getTime());
+  }
+
+  @After public void tearDown() throws Exception {
+    deleteFiles();
+  }
+
+  private void deleteFiles() {
+    File currentDirectory = new File(".");
```

Comment:
<!--new_thread; commit:c6c7070ad92424707d3720d3a4dc2c0fb6961440; resolved:0-->
You can use FileSystems abstraction to delete these files in a file-system agnostic manner.

You'll have to do this from a DoFn to make sure that all registered FileSystems are available for deletion. Also, you'll have to make sure that files are read before deletion.

One way to enforce this will be to pass the 'consolidatedHashcode' as a side input to a ParDo() that performs deletion. The main input to this ParDo should be the PCollection that has the set of file names (result of 'Write content to files' step).

Reply:
Ok, I'll check that API.

Reply:
I think I've found a way to use the `FileSystems API` and stay in the `tearDown()` method. This way I: 
 - do not have to mess with the pipeline code and delete inside the @Test method
 - files allways get deleted - no matter what happens with the pipeline (exceptions etc)
 - do not have to wait for the files to be read (this doesn't concern me in the `tearDown()` method)
 - stay "filesytem agnositic" because this is what FileSystems API is for

Could you review once more and validate my idea?